Makefile.shared

```diff
@@ -311,3 +311,30 @@ partialclean::
 	rm -f driver/compdynlink.mli
 	rm -f driver/compdynlink.mlopt
 
+# The native toplevel
+
+compilerlibs/ocamlopttoplevel.cmxa: $(OPTTOPLEVEL:.cmo=.cmx)
+	$(CAMLOPT) -a -o $@ $(OPTTOPLEVEL:.cmo=.cmx)
+partialclean::
+	rm -f compilerlibs/ocamlopttoplevel.cmxa
+
+# When the native toplevel executable has an extension (e.g. ".exe"),
+# provide a phony 'ocamlnat' synonym
+
+ifneq ($(EXE),)
+.PHYNY: ocamlnat
```

Comment:
Alain Frisch (2016/07/31 07:20 -0700):
> > +ifneq ($(EXE),)
> > +.PHYNY: ocamlnat
> 
> Please check, but I believe there is some magic in Cygwin's make which
> makes this useless.

I just tried commenting the rule. When it is commented, make ocamlnat
fails with the message

make: *** No rule to make target 'ocamlnat'.  Stop.